packages/manager/apps/web/client/app/hosting/cdn/flush/hosting-cdn-flush.controller.js

```diff
@@ -38,26 +29,31 @@ export default class HostingFlushCdnCtrl {
   }
 
   flushCdn() {
+    this.loading = true;
     const flushPromise = this.isV1CDN
       ? this.flushV1CDN()
       : this.flushSharedCDN();
     return flushPromise
       .then(() => {
-        this.Alerter.success(
-          this.$translate.instant('hosting_dashboard_cdn_flush_success'),
-          this.$scope.alerts.main,
-        );
-        this.$rootScope.$broadcast('hosting.cdn.flush.refresh');
+        this.onFlushSuccess();
+        return this.goBack().then(() => {
+          this.Alerter.success(
+            this.$translate.instant('hosting_dashboard_cdn_flush_success'),
+            'app.alerts.main',
```

Comment:
Should the controller be aware of the parent structure ?

Reply:
Hmm you suggest to pass thie logic into the routing ?

Reply:
No but it refers to the name alert is given in the parent right ? 
What has been done in other places is that the goBack only takes a message as parameter so it's the function handling how and where the message is displayed.
In your case if we rename `app.alerts.main` we have to patch all children using `goBack`

Reply:
Ok sure, I've just change the existing code. Now apply, thanks.